### PR TITLE
refactor(events): Tenant config in API, Connector and Outgoing Web-hook events

### PR DESCRIPTION
**Pull request by ShivanshMathurJuspay**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

The changes are required for pushing tenant ID for API events, Connector events and Outgoing Webhook events

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Attaching screenshots 

Screenshot 2024-12-09 at 3 20 34 PM
Screenshot 2024-12-09 at 3 20 08 PM
Screenshot 2024-12-09 at 3 20 21 PM

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/core/webhooks/incoming.rs [hunk 1]
```diff
@@ -99,6 +99,7 @@ pub async fn incoming_webhooks_wrapper<W: types::OutgoingWebhookType>(
         .attach_printable("Could not convert webhook effect to string")?;
 
     let api_event = ApiEvent::new(
+        state.tenant.tenant_id.clone(),
```
Comment (NishantJoshi00) [inline crates/router/src/core/webhooks/incoming.rs:102]: If we are only logging this event by passing a reference in `log_event` function. Why do we need to clone?
(Just minor optimization, you can pick this up later if you wish)

**Review by NishantJoshi00 — DISMISSED** [review]

**Review by jarnura — APPROVED** [review]
Remove clone() will be good, same data clone for every event will be more costly only

**Comment by tsdk02** [comment]
The `tenant_id` needs to be passed in the v2 code too for `api_events`.
A required check is failing due to this.

**Review by tsdk02 — APPROVED** [review]

File: crates/router/src/core/webhooks/incoming_v2.rs [hunk 2]
```diff
@@ -91,6 +91,8 @@ pub async fn incoming_webhooks_wrapper<W: types::OutgoingWebhookType>(
         .attach_printable("Could not convert webhook effect to string")?;
 
     let api_event = ApiEvent::new(
+        #[cfg(feature = "v1")]
```
Comment (Narayanbhat166) [inline crates/router/src/core/webhooks/incoming_v2.rs:94]: why do we have a v1 feature flag in a module which is for v2 only

File: crates/router/src/events/api_logs.rs [hunk 3]
```diff
@@ -47,6 +49,7 @@ pub struct ApiEvent {
 impl ApiEvent {
     #[allow(clippy::too_many_arguments)]
     pub fn new(
+        #[cfg(feature = "v1")] tenant_id: common_utils::id_type::TenantId,
```
Comment (Narayanbhat166) [inline crates/router/src/events/api_logs.rs:52]: why do we have this only for v1? can we have this for v2 as well?

**Review by jarnura — APPROVED** [review]

**Review by tsdk02 — APPROVED** [review]

**Pull request merged**